pytensor/tensor/elemwise.py

```diff
@@ -419,45 +419,67 @@ def get_output_info(self, dim_shuffle, *inputs):
         # of all inputs in parallel... the all() gives us each output
         # broadcastable bit in turn.
 
-        def get_most_specialized_shape(shapes):
-            shapes = set(shapes)
-            # All shapes are the same
-            if len(shapes) == 1:
-                return tuple(shapes)[0]
-
-            # Only valid indeterminate case
-            if shapes == {None, 1}:
-                return None
-
-            shapes.discard(1)
-            shapes.discard(None)
-            if len(shapes) > 1:
-                raise ValueError
-            return tuple(shapes)[0]
+        def get_most_specialized_shape(shapes, bcasting):
+            seen_dims = set(zip(shapes, bcasting))
+            if len(seen_dims) == 1:
+                return next(iter(seen_dims))[0]
```

Comment:
Small suggestion, since the return logic is always the same why not do it once at the end of the function (and make the last branch an `else` branch)

Reply:
NVM this looks a bit more readable